driver/config/tls.go

```diff
@@ -0,0 +1,70 @@
+package config
+
+import (
+	"crypto/tls"
+
+	"github.com/ory/x/tlsx"
+)
+
+const (
+	KeySuffixTLSStrict               = "tls.strict"
+	KeySuffixTLSAllowTerminationFrom = "tls.allow_termination_from"
+	KeySuffixTLSCertString           = "tls.cert.base64"
+	KeySuffixTLSKeyString            = "tls.key.base64"
+	KeySuffixTLSCertPath             = "tls.cert.path"
+	KeySuffixTLSKeyPath              = "tls.key.path"
+
+	KeyTLSAllowTerminationFrom = "serve." + KeySuffixTLSAllowTerminationFrom
+	KeyTLSCertString           = "serve." + KeySuffixTLSCertString
+	KeyTLSKeyString            = "serve." + KeySuffixTLSKeyString
+	KeyTLSCertPath             = "serve." + KeySuffixTLSCertPath
+	KeyTLSKeyPath              = "serve." + KeySuffixTLSKeyPath
+)
+
+type TLSConfig interface {
+	Strict() bool
+	AllowTerminationFrom() []string
+	Certificate() ([]tls.Certificate, error)
+}
+
+func (p *Provider) TLS(iface ServeInterface) TLSConfig {
+	strict := false
+	if !p.forcedHTTP() {
+		strict = p.p.Bool(iface.Key(KeySuffixTLSStrict))
```

Comment:
I'm not entirely sure what the new `tls.strict` is for? Is there a reason why we need it on top of `--dangerous-force-http`? Also, doesn't that mean that we need to set `strict` to `false` (default is `true`) for `--dangerous-force-http` to have an effect?

Reply:
This code should be probably improved, it just makes it respect dangerous-force-http flag over tls.strict config param.

Reply:
Yeah makes sense, I misread what this is doing here! This is ok :)